### PR TITLE
CORE-13980 Membership metrics

### PR DESCRIPTION
**Pull request by charlieR3**
Enable the capture of membership metrics for:
* Member list cache size
* Registration stage timing
* Synchronisation stage timing
* Member info distribution timing

This also groups the membership metrics under one object.

Dashboard added here: https://github.com/corda/engineering-helm/pull/42
That PR includes links to sample dashboards showing these metrics.

### CODE REVIEW

**Review by dimosr — COMMENTED** [review]
LGTM - just some detekt failures that need fixing

**Review by williamvigorr3 — DISMISSED** [review]
LGTM

**Review by JamesHR3 — DISMISSED** [review]

**Review by JamesHR3 — APPROVED** [review]

File: components/membership/membership-group-read-impl/src/main/kotlin/net/corda/membership/impl/read/cache/MemberListCache.kt [hunk 1]
```diff
@@ -30,28 +33,43 @@ interface MemberListCache : MemberDataListCache<MemberInfo> {
...
             cache.clear()
         }
 
+        /**
+         * Returns the metric for setting the current member list cache size. We need to hold a reference to the
+         * [SettableGauge] so that the cached gauge metric in the micrometer library is referencing the expected value.
+         * If we recreated the settable gauge, the value referenced by the cached gauge will be cleared by the garbage
+         * collector and NaN will be reported in the metric for cache size.
+         */
+        private fun getCacheSizeMetric(holdingId: HoldingIdentity) = cacheSizeMetrics
```
Comment (driessamyn) [inline components/membership/membership-group-read-impl/src/main/kotlin/net/corda/membership/impl/read/cache/MemberListCache.kt:68]: Do we need this?
The caching metrics are already exposed directly from Caffeine.

Comment (charlieR3) [reply components/membership/membership-group-read-impl/src/main/kotlin/net/corda/membership/impl/read/cache/MemberListCache.kt:68]: Maybe cache isn't the best word here. It's an in-memory map and not a caffeine cache

**Review by driessamyn — APPROVED** [review]
LGTM

**Pull request merged**